video/out/drm_atomic.c

```diff
@@ -458,3 +497,113 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
 
     return (ret == 0);
 }
+
+static inline uint16_t clamp16(float x) {
+    return (uint16_t)MPCLAMP(x, 0.0, 65535.0);
+}
+
+static inline uint16_t clamp16_round(float x) {
+    return clamp16(roundf(x * 50000.0));
+}
+
+void drm_send_hdrmeta(struct drm_atomic_context *ctx, struct mp_colorspace *color) {
+#if HAVE_DRM_HDR
+
+    uint32_t property_hdr_id;
+    if (0 > drm_object_get_property_id(ctx->connector, "HDR_OUTPUT_METADATA", &property_hdr_id))
+        return;
+
+    struct drm_hdr_metadata *p = &ctx->hdr_metadata;
```

Comment:
I don't see why this needs to be part of a public struct, it never lives longer than this function's scope anyway.

Reply:
Eventually the goal will be to cache the metadata, and reuse it if it doesn't change. I need to do more research on the drm APIs to see if there's a better way to do this, anyway.

Reply:
I don't think that's necessary even then, you will need to do the modification test on the _input_ data